src/main/java/bio/terra/cli/command/resources/create/GcsBucket.java

```diff
@@ -0,0 +1,83 @@
+package bio.terra.cli.command.resources.create;
+
+import bio.terra.cli.command.helperclasses.BaseCommand;
+import bio.terra.cli.command.helperclasses.FormatOption;
+import bio.terra.cli.service.WorkspaceManager;
+import bio.terra.workspace.model.AccessScope;
+import bio.terra.workspace.model.CloningInstructionsEnum;
+import bio.terra.workspace.model.GcpGcsBucketDefaultStorageClass;
+import bio.terra.workspace.model.ResourceDescription;
+import java.util.Collections;
+import picocli.CommandLine;
+
+/** This class corresponds to the fourth-level "terra resources create gcs-bucket" command. */
+@CommandLine.Command(
+    name = "gcs-bucket",
+    description = "Add a controlled GCS bucket.",
+    showDefaultValues = true)
+public class GcsBucket extends BaseCommand {
+  @CommandLine.Option(
+      names = "--name",
+      required = true,
+      description =
+          "Name of the resource, scoped to the workspace. Only alphanumeric and underscore characters are permitted.")
+  private String name;
+
+  @CommandLine.Option(names = "--description", description = "Description of the resource")
+  private String description;
+
+  @CommandLine.Option(
+      names = "--cloning",
+      description =
+          "Instructions for handling when cloning the workspace: ${COMPLETION-CANDIDATES}")
+  private CloningInstructionsEnum cloning = CloningInstructionsEnum.NOTHING;
+
+  @CommandLine.Option(
+      names = "--access",
+      description = "Access scope for the resource: ${COMPLETION-CANDIDATES}")
+  private AccessScope access = AccessScope.SHARED_ACCESS;
+
+  @CommandLine.Option(
+      names = "--bucket-name",
+      required = true,
+      description =
+          "Name of the GCS bucket, without the prefix. (e.g. 'my-bucket', not 'gs://my-bucket')")
+  private String bucketName;
+
+  @CommandLine.Option(
+      names = "--storage",
+      description =
+          "Storage class (https://cloud.google.com/storage/docs/storage-classes): ${COMPLETION-CANDIDATES}")
+  private GcpGcsBucketDefaultStorageClass storageClass;
+
+  @CommandLine.Option(
+      names = "--location",
+      description = "Bucket location (https://cloud.google.com/storage/docs/locations)")
+  private String location;
+
+  @CommandLine.Mixin FormatOption formatOption;
+
+  /** Add a controlled GCS bucket to the workspace. */
+  @Override
+  protected void execute() {
+    // TODO (PF-486): allow the user to specify lifecycle rules on the bucket
+    ResourceDescription resource =
+        new WorkspaceManager(globalContext, workspaceContext)
+            .createControlledGcsBucket(
+                name,
+                description,
+                cloning,
+                access,
+                bucketName,
+                storageClass,
+                Collections.emptyList(),
+                location);
```

Comment:
It might be nice to allow specifying a json file with the creation parameters. But that seems very speculative, so probably best just to keep it in mind as a possible future option.

Reply:
Agreed that might be coming. For the ticket I'm working on now, `terra resources create gcs-bucket` will optionally take a JSON file that specifies the lifecycle rules, the same as the `gsutil lifecycle set` command [does](https://cloud.google.com/storage/docs/gsutil/commands/lifecycle). I'm planning to use the same JSON [format](https://cloud.google.com/storage/docs/json_api/v1/buckets#resource) as `gsutil`. So if we ever want to allow passing a JSON file for the whole resource definition (not just lifecycle rules), perhaps we can follow the same approach.